XRTK-Core/Packages/com.xrtk.core/Services/InputSystem/MixedRealityInputSystem.cs

```diff
@@ -90,10 +91,14 @@ public MixedRealityInputSystem(MixedRealityInputSystemProfile profile) : base(pr
         private InputEventData<Vector3> positionInputEventData;
         private InputEventData<Quaternion> rotationInputEventData;
         private InputEventData<MixedRealityPose> poseInputEventData;
+        private InputEventData<IDictionary<TrackedHandJoint, MixedRealityPose>> handJointPoseInputEventData;
 
         private SpeechEventData speechEventData;
         private DictationEventData dictationEventData;
 
+        private InputEventData<HandMeshUpdatedEventData> handMeshUpdatedEventData;
```

Comment:
These hand specific events don't need to be added to the input system.

Lets revisit them as gestures.

Reply:
They should be a part of a hand service. Imo

Reply:
Controllers (which hands are) should only be comprised of three main parts.  The Data Provider, The controller mapper (if applicable) and the visualizer. (excluding base definitions for structs and such)

So hands should not be a separate service, they start as a data provider,  allowing different providers to be used to get data from different sources, e.g. Leap Motion / ML / HL2.

Reply:
I disagree @SimonDarksideJ.
If you never need to make use of the hand joints for writing interactions, then that's fine. However, if you do, you need an easy way to access hand joint data without doing a ton of casting.

A service allows you to know if a given hand is tracked and know the pos/rot of any given joint, is essential, in my opinion. The visualizer can be a part of this, but a visualizer on it's own isn't that useful.

Reply:
From another comment, it seems that the data provider is something that is generic and accessible? 

If so then I don't disagree. I just assumed it was something that provided data from a platform level.

I just want it to be the same provider whether you're using leap, magic leap or HL2.

Reply:
As of right now the `IMixedRealityHand` has a contract for `TryGetJoint()`. Is this sufficient or do you think we need a service to facilitate more data flow @provencher?

Reply:
I think it also needs IsTracked and IsInPointing pose, to show it's ready to be used as a hand ray pointer.

I like the idea of having a central authority to grab the hand data, and get the hand controllers, but that should be sufficient.